### ISSUE TITLE
Support lowering mhlo.select_and_scatter to linalg

### ISSUE THREAD

**Issue by asaadaldien** [google/iree#5553]
This is required to run gradients of pooling ops as in the example:

```python
def noop_pool(x):
  return nn.max_pool(x, (1, 1), (1, 1))

x = jnp.ones((2, 2, 2))
iree.jax.jit(grad_sum(noop_pool))(x)
```

```mlir
// noop_pool.mhlo.mlir
module  {
  func @main(%arg0: tensor<2x2x2xf32>) -> tensor<2x2x2xf32> attributes {iree.module.export} {
    %0 = mhlo.constant dense<false> : tensor<i1>
    %1 = "mhlo.broadcast_in_dim"(%arg0) {broadcast_dimensions = dense<[1, 2, 3]> : tensor<3xi64>} : (tensor<2x2x2xf32>) -> tensor<1x2x2x2xf32>
    %2 = mhlo.constant dense<0xFF800000> : tensor<f32>
    %3 = "mhlo.reduce_window"(%1, %2) ( {
    ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):  // no predecessors
      %17 = mhlo.maximum %arg1, %arg2 : tensor<f32>
      "mhlo.return"(%17) : (tensor<f32>) -> ()
    }) {base_dilations = dense<1> : tensor<4xi64>, padding = dense<0> : tensor<4x2xi64>, window_dilations = dense<1> : tensor<4xi64>, window_dimensions = dense<1> : tensor<4xi64>, window_strides = dense<1> : tensor<4xi64>} : (tensor<1x2x2x2xf32>, tensor<f32>) -> tensor<1x2x2x2xf32>
    %4 = "mhlo.reshape"(%3) : (tensor<1x2x2x2xf32>) -> tensor<2x2x2xf32>
    %5 = mhlo.constant dense<0.000000e+00> : tensor<f32>
    %6 = "mhlo.reduce"(%4, %5) ( {
    ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):  // no predecessors
      %17 = mhlo.add %arg1, %arg2 : tensor<f32>
      "mhlo.return"(%17) : (tensor<f32>) -> ()
    }) {dimensions = dense<[0, 1, 2]> : tensor<3xi64>} : (tensor<2x2x2xf32>, tensor<f32>) -> tensor<f32>
    %7 = mhlo.constant dense<1.000000e+00> : tensor<f32>
    %8 = "mhlo.broadcast_in_dim"(%7) {broadcast_dimensions = dense<> : tensor<0xi64>} : (tensor<f32>) -> tensor<2x2x2xf32>
    %9 = "mhlo.broadcast_in_dim"(%8) {broadcast_dimensions = dense<[1, 2, 3]> : tensor<3xi64>} : (tensor<2x2x2xf32>) -> tensor<1x2x2x2xf32>
    %10 = mhlo.constant dense<0.000000e+00> : tensor<f32>
    %11 = "mhlo.select_and_scatter"(%1, %9, %10) ( {
    ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):  // no predecessors
      %17 = "mhlo.compare"(%arg1, %arg2) {comparison_direction = "GE"} : (tensor<f32>, tensor<f32>) -> tensor<i1>
      "mhlo.return"(%17) : (tensor<i1>) -> ()
    },  {
    ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):  // no predecessors
      %17 = mhlo.add %arg1, %arg2 : tensor<f32>
      "mhlo.return"(%17) : (tensor<f32>) -> ()
    }) {padding = dense<0> : tensor<4x2xi64>, window_dimensions = dense<1> : tensor<4xi64>, window_strides = dense<1> : tensor<4xi64>} : (tensor<1x2x2x2xf32>, tensor<1x2x2x2xf32>, tensor<f32>) -> tensor<1x2x2x2xf32>
    %12 = mhlo.constant dense<0.000000e+00> : tensor<f32>
    %13 = "mhlo.reduce"(%11, %12) ( {
    ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):  // no predecessors
      %17 = mhlo.add %arg1, %arg2 : tensor<f32>
      "mhlo.return"(%17) : (tensor<f32>) -> ()
    }) {dimensions = dense<0> : tensor<1xi64>} : (tensor<1x2x2x2xf32>, tensor<f32>) -> tensor<2x2x2xf32>
    %14 = "mhlo.broadcast_in_dim"(%13) {broadcast_dimensions = dense<[0, 1, 2]> : tensor<3xi64>} : (tensor<2x2x2xf32>) -> tensor<2x2x2xf32>
    %15 = "mhlo.tuple"(%14) : (tensor<2x2x2xf32>) -> tuple<tensor<2x2x2xf32>>
    %16 = "mhlo.get_tuple_element"(%15) {index = 0 : i32} : (tuple<tensor<2x2x2xf32>>) -> tensor<2x2x2xf32>
    return %16 : tensor<2x2x2xf32>
  }
}
```

**Comment by asaadaldien** [comment]
https://www.tensorflow.org/xla/operation_semantics#selectandscatter

**Comment by MaheshRavishankar** [comment]
Not sure if this is still needed. If needed there are scatter operations supported in IREE. If there are other issues, please file a separate bug for this.
